### PR TITLE
Added usage of JPARepository and added logic for search() method usin…

### PR DESCRIPTION
**Pull request by holsoni**
…g CriteriaQuery

### CODE REVIEW

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
 @RestController
```
Comment (keda-vlad) [inline src/main/java/com/example/bookstore/controller/BookController.java:22]: Book Endpoints:
GET: /api/books (Retrieve book catalog) (should be done in the previous PR)
GET: /api/books/{id} (Retrieve book details) (should be done in the previous PR)
POST: /api/books (Create a new book) (should be done in the previous PR)
PUT: /api/books/{id} (Update a specific book)  <------------------------------------ u miss this

**Review by keda-vlad — CHANGES_REQUESTED** [review]
Nice, but only one thing)

**Review by keda-vlad — APPROVED** [review]
Nice)

Optional, maybe better to create constant for "author" in AuthorSpecificationProvider  and in BookSpecificationBuilder

File: src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -49,11 +51,20 @@ public void deleteById(Long id) {
         bookRepository.deleteById(id);
     }
 
+    @Override
+    public BookDto updateById(Long id, UpdateBookRequestDto updateBookRequestDto) {
+        Optional<Book> fromDb = bookRepository.findById(id);
```
Comment (keda-vlad) [inline src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java]: Do u need it?

File: src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -39,4 +44,25 @@ public BookDto findById(Long id) {
...
         return bookMapper.toDto(book);
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, UpdateBookRequestDto updateBookRequestDto) {
```
Comment (vovasalyha) [inline src/main/java/com/example/bookstore/service/impl/BookServiceImpl.java:54]: Fetching an entity before update is usually safer.

**Review by vovasalyha — APPROVED** [review]
GJ

**Pull request merged**